src/main/java/com/jw/resources/OrderResource.java

```diff
@@ -53,6 +57,7 @@ public OrderResponse saveOrder(@Valid OrderRequest orderRequest) {
 
     @DELETE
     @Path("/{id}")
+    //REVIEW-VINI: Should this method be present on FinalizeOrderResource? IMHO delete order is also a way to finalize an order
```

Comment:
In that case, deleting is slightly different than finalizing an order. 
Finalizing:  changing Order status to finalized, and decreasing reserved products. In the result, they "disappear" because they were bought. Order still appear in the database.
Deleting: each reserved product goes back to the available amount, because client resigned from buying. Since order is deleted, it is removed from database